app/controllers/provider_interface/offer/conditions_controller.rb

```diff
@@ -24,31 +16,75 @@ def edit
       end
 
       def update
-        @wizard = OfferWizard.new(offer_store, attributes_for_wizard)
+        handle_form_input(action: :edit)
+      end
 
-        if @wizard.valid_for_current_step?
-          @wizard.save_state!
+    private
+
+      def handle_form_input(action:)
+        @wizard = OfferWizard.new(offer_store, attributes_for_wizard)
 
-          redirect_to [:edit, :provider_interface, @application_choice, :offer, @wizard.next_step]
+        if add_another_condition?
+          add_empty_condition
+          redirect_to action: action, anchor: anchor_for_further_condition
+        elsif remove_condition_param.present?
+          remove_condition(remove_condition_param)
+          redirect_to action: action, anchor: anchor_for_further_condition
         else
-          render :edit
+          submit_form(action: action)
         end
       end
 
-    private
+      def remove_condition_param
+        params[:remove_condition]
+      end
 
-      def conditions_params
-        params.require(:provider_interface_offer_wizard)
-              .permit(further_conditions: {}, standard_conditions: [])
+      def add_another_condition?
+        params[:commit] == 'add_another_condition'
       end
 
       def attributes_for_wizard
         attributes = conditions_params
 
-        further_conditions = conditions_params.to_h['further_conditions'].values.map { |hash| hash['text'] }
+        further_conditions = conditions_params.fetch('further_conditions', {}).values.map { |hash| hash['text'] }
 
         attributes.merge!(further_conditions: further_conditions, current_step: 'conditions')
       end
+
+      def conditions_params
+        params.require(:provider_interface_offer_wizard)
+              .permit(further_conditions: {}, standard_conditions: [])
+      end
+
+      def add_empty_condition
```

Comment:
this logic feels more suited in the wizard, which should decide what we do next?